sonic-stormond/scripts/stormond

```diff
@@ -0,0 +1,379 @@
+        
+        return fsio_dict["total_fsio_reads"], fsio_dict["total_fsio_writes"]
+
+
+
+    # Update the Storage device info to State DB
+    def update_storage_info_status_db(self, disk_device, disk_fields, kvp_dict):
+
+        for field in disk_fields:
+            fvp = swsscommon.FieldValuePairs([(field, str(kvp_dict[field]))])
+            self.device_table.set(disk_device, fvp)
+
+
+    # Get Static attributes and update the State DB, once
+    def get_static_fields(self):
+
+        # Get relevant information about each Storage Device on the switch
+        for storage_device in self.storage.devices:
+            try:
+                # Unlikely scenario
+                if self.storage.devices[storage_device] is None:
+                    self.log_info("{} does not have an instantiated object. Static Information cannot be gathered.".format(storage_device))
+                    continue
+
+                static_kvp_dict = dict((field, "") for field in self.static_fields)
+
+                static_kvp_dict["device_model"] = self.storage.devices[storage_device].get_model()
+                static_kvp_dict["serial"] = self.storage.devices[storage_device].get_serial()
+
+                self.log_info("Storage Device: {}, Device Model: {}, Serial: {}".format(storage_device, static_kvp_dict["device_model"], static_kvp_dict["serial"]))
+
+                # update Storage Device Status to DB
+                self.update_storage_info_status_db(storage_device, self.static_fields, static_kvp_dict)
+
+            except Exception as ex:
+                self.log_info("get_static_fields() failed with: {}".format(str(ex)))
+
+    # Get Dynamic attributes and update the State DB
+    def get_dynamic_fields(self):
+
+        # Get relevant information about each storage disk on the device
+        for storage_device in self.storage.devices:
+            try:
+                if self.storage.devices[storage_device] is None:
+                    self.log_info("Storage device '{}' does not have an instantiated object. Dynamic Information cannot be gathered.".format(storage_device))
+                    continue
+
+                dynamic_kvp_dict = dict((field, "") for field in self.dynamic_fields)
+
+                dynamic_kvp_dict["firmware"] = self.storage.devices[storage_device].get_firmware()
+                dynamic_kvp_dict["health"] = self.storage.devices[storage_device].get_health()
+                dynamic_kvp_dict["temperature"] = self.storage.devices[storage_device].get_temperature()
+                dynamic_kvp_dict["latest_fsio_reads"] = self.storage.devices[storage_device].get_fs_io_reads()
+                dynamic_kvp_dict["latest_fsio_writes"] = self.storage.devices[storage_device].get_fs_io_writes()
+                dynamic_kvp_dict["disk_io_reads"] = self.storage.devices[storage_device].get_disk_io_reads()
```

Comment:
This API parses the value that's fetched during object instantiation. For dynamic update we should also fetch before parsing but currently we do fetch only once during the init. Can you please update the logic?

Reply:
As discussed, I have added an API to fetch and parse vendor data in SsdUtil. I will make a call to this API on every get_dynamic_fields() invocation. (EmmcUtil does not have this issue.)

Reply:
Fixed in latest commit